cal/cal/models.py

```diff
@@ -711,6 +710,44 @@ def get_time_series(self, timezone='UTC', time_step='week', calendar_ids=None, s
         return get_time_series(self, timezone, time_step, calendar_ids, start, end)
 
 
+    def get_category_stats(self, categories, start=None, end=None):
+        """
+        Returns a list of category-hour tuples corresponding to the number of hours per category
+        that corresponds to this Tag.
+        """
+        category_hours = []
+        keywords = self.keywords.split(',')
+
+        for category in categories:
+            querysets = [
+                    GEvent.objects
+                    .filter(calendar__user=category.user, calendar=category.calendar,
+                            name__regex=r'\b(?i)[^a-zA-Z\d:]?'+keyword+r'(?i)[^a-zA-Z\d:]?\b',
+                            color_index=category.color_index)
+                    for keyword in keywords
+                    ]
+
+            events_qs = reduce(lambda qs1, qs2: qs1 | qs2, querysets)
+            events_qs = events_qs.order_by('start')
+
+            if start:
+                events_qs = events_qs.filter(end__gt=start)
+            if end:
+                events_qs = events_qs.filter(start__lt=end)
+            else:
+                events_qs = events_qs.filter(start__lt=datetime.now(pytz.utc))
+
+            hours = round(EventCollection(lambda:[e for e in events_qs]).total_time() / 3600.0, 2)
```

Comment:
`EventCollection` needs to have a `set` passed in. We're not currently using the intersects/unions yet but this will save us time in the future when we do.

Brief overview of levels of abstraction:
`query`: always returns a `QuerySet` for database level joins/intersections
`get_events`: always returns a `set` for Python level joins/intersections